enterprise/models.py

```diff
@@ -415,14 +426,27 @@ def get_course_run_enrollment_url(self, course_run_key):
         Returns:
             (str): Enterprise landing page url.
         """
-        url = urljoin(
-            get_configuration_value('LMS_ROOT_URL', settings.LMS_ROOT_URL),
-            reverse(
-                'enterprise_course_run_enrollment_page',
-                kwargs={'enterprise_uuid': self.uuid, 'course_id': course_run_key}
+        parent_course_key = get_course_key_from_course_run_key(course_run_key)
+        params = utils.get_enterprise_utm_context(self)
+        if self.enable_learner_portal:
+            params.update({'course_run_key': course_run_key})
+            url = '{}/{}/course/{}'.format(
+                get_configuration_value(
+                    'ENTERPRISE_LEARNER_PORTAL_BASE_URL',
+                    settings.ENTERPRISE_LEARNER_PORTAL_BASE_URL
+                ),
+                self.slug,
+                parent_course_key,
             )
-        )
-        return utils.update_query_parameters(url, utils.get_enterprise_utm_context(self))
+        else:
+            url = urljoin(
+                get_configuration_value('LMS_ROOT_URL', settings.LMS_ROOT_URL),
+                reverse(
+                    'enterprise_course_run_enrollment_page',
+                    kwargs={'enterprise_uuid': self.uuid, 'course_id': course_run_key}
+                )
+            )
+        return utils.update_query_parameters(url, params)
```

Comment:
Nit: this chunk of code is almost identical to the above (save for `course_key` vs. `course_run_key`).  Maybe you want a couple of helper functions like `_learner_portal_enrollment_url(content_key)` and `_enterprise_enrollment_url(content_key)`.